disnake/channel.py

```diff
@@ -1817,6 +1854,461 @@ class NewsChannel(TextChannel):
+        reason: Optional[:class:`str`]
+            The reason for creating the thread. Shows up on the audit log.
+
+        Raises
+        ------
+        Forbidden
+            You do not have permissions to create a thread.
+        HTTPException
+            Starting the thread failed.
+        TypeError
+            Specified both ``file`` and ``files``,
+            or you specified both ``embed`` and ``embeds``,
+            or you specified both ``view`` and ``components``.
+        InvalidArgument
+            Specified more than 10 embeds,
+            or more than 10 files,
+            or you have passed an object that is not :class:`File`.
+
+        Returns
+        -------
+        :class:`Thread`
+            The newly created thread.
+        """
+        from .webhook.async_ import handle_message_parameters_dict
+
+        params = handle_message_parameters_dict(
+            content,
+            embed=embed,
+            embeds=embeds,
+            file=file,
+            files=files,
+            view=view,
+            components=components,
+            allowed_mentions=allowed_mentions,
+            stickers=stickers,
+        )
+
+        if auto_archive_duration is not None:
+            auto_archive_duration = cast(
+                "ThreadArchiveDurationLiteral", try_enum_to_int(auto_archive_duration)
+            )
+
+        if params.files and len(params.files) > 10:
+            raise InvalidArgument("files parameter must be a list of up to 10 elements")
+        elif params.files and not all(isinstance(file, File) for file in params.files):
+            raise InvalidArgument("files parameter must be a list of File")
+
+        try:
+            thread_data = await self._state.http.start_thread_in_forum_channel(
+                self.id,
+                name=name,
+                auto_archive_duration=auto_archive_duration or self.default_auto_archive_duration,
+                rate_limit_per_user=slowmode_delay or 0,
+                type=ChannelType.public_thread.value,
+                files=params.files,
+                reason=reason,
+                **params.payload,
+            )
+        finally:
+            if params.files:
+                for f in params.files:
+                    f.close()
+
+        if view:
+            self._state.store_view(view, int(thread_data["id"]))
+
+        return Thread(guild=self.guild, data=thread_data, state=self._state)
+
+    def archived_threads(
+        self,
+        *,
+        limit: Optional[int] = 50,
+        before: Optional[Union[Snowflake, datetime.datetime]] = None,
+    ) -> ArchivedThreadIterator:
+        """Returns an :class:`~disnake.AsyncIterator` that iterates over all archived threads in the channel.
+
+        You must have :attr:`~Permissions.read_message_history` permission to use this.
+
+        Parameters
+        ----------
+        limit: Optional[:class:`bool`]
```

Comment:
```suggestion
        limit: Optional[:class:`int`]
```
This is also wrong in TextChannel

Reply:
we'll update the ForumChannel docstring in this pr, @shiftinv, could you update it in #470?